service/rttests/rt_test.go

```diff
@@ -0,0 +1,390 @@
+		Values: []string{
+			"red",
+			"green",
+			"blue",
+		},
+	})
+	if err != nil {
+		if returnStatus, ok := status.FromError(err); ok && returnStatus.Code() == codes.AlreadyExists {
+			slog.Info("attribute already exists")
+		} else {
+			slog.Error("could not create attribute", slog.String("error", err.Error()))
+			return err
+		}
+	} else {
+		slog.Info("attribute created")
+	}
+
+	slog.Info("creating attribute cards with hierarchy rule")
+	_, err = s.Attributes.CreateAttribute(context.Background(), &attributes.CreateAttributeRequest{
+		Name:        "cards",
+		NamespaceId: exampleNamespace.GetId(),
+		Rule:        *policy.AttributeRuleTypeEnum_ATTRIBUTE_RULE_TYPE_ENUM_HIERARCHY.Enum(),
+		Values: []string{
+			"king",
+			"queen",
+			"jack",
+		},
+	})
+	if err != nil {
+		if returnStatus, ok := status.FromError(err); ok && returnStatus.Code() == codes.AlreadyExists {
+			slog.Info("attribute already exists")
+		} else {
+			slog.Error("could not create attribute", slog.String("error", err.Error()))
+			return err
+		}
+	} else {
+		slog.Info("attribute created")
+	}
+
+	slog.Info("##################################\n#######################################")
+
+	allAttr, err := s.Attributes.ListAttributes(context.Background(), &attributes.ListAttributesRequest{})
+	if err != nil {
+		slog.Error("could not list attributes", slog.String("error", err.Error()))
+		return err
+	}
+	slog.Info(fmt.Sprintf("list attributes response: %s", protojson.Format(allAttr)))
+
+	slog.Info("##################################\n#######################################")
+
+	// get the attribute ids for the values were mapping to the client
+	var attributeValueIDs []string
+	fqnResp, err := s.Attributes.GetAttributeValuesByFqns(context.Background(), &attributes.GetAttributeValuesByFqnsRequest{
+		Fqns:      attributesToMap,
+		WithValue: &policy.AttributeValueSelector{},
+	})
+	if err != nil {
+		slog.Error("get attribute values by fqn ", slog.String("error", err.Error()))
+		return err
+	}
+	for _, attribute := range attributesToMap {
+		attributeValueIDs = append(attributeValueIDs, fqnResp.GetFqnAttributeValues()[attribute].GetValue().GetId())
+	}
+
+	// create subject mappings
+	slog.Info("creating subject mappings for client " + testConfig.ClientID)
+	for _, attributeID := range attributeValueIDs {
+		_, err = s.SubjectMapping.CreateSubjectMapping(context.Background(), &subjectmapping.CreateSubjectMappingRequest{
+			AttributeValueId: attributeID,
+			Actions: []*policy.Action{{Value: &policy.Action_Standard{
+				Standard: policy.Action_STANDARD_ACTION_DECRYPT,
+			}},
+				{Value: &policy.Action_Standard{
+					Standard: policy.Action_STANDARD_ACTION_TRANSMIT,
+				}},
+			},
+			NewSubjectConditionSet: &subjectmapping.SubjectConditionSetCreate{
+				SubjectSets: []*policy.SubjectSet{
+					{ConditionGroups: []*policy.ConditionGroup{
+						{Conditions: []*policy.Condition{{
+							SubjectExternalSelectorValue: ".clientId",
```

Comment:
is this a working subjectmapping. I have something like this, but i had to do "\".clientId\"". seems like it needs to be quoted due to here https://github.com/opentdf/platform/blame/41ca82d692209d120bfa52800fa0988bf373b0b5/service/internal/jqbuiltin/jq_builtin.go#L90

Reply:
it works without the additional quotes, i believe that unescape query function is intended to remove extra sets of quotes if present

Reply:
ahh, you know what i figured out what happened. my mistake fixed two things at once, and had the wrong conclusion.